### PR TITLE
Upgrade nightly wheels to rocm5.4.2

### PR DESCRIPTION
**Pull request by BLOrange-AMD**

### CODE REVIEW

**Review by jithunnair-amd — APPROVED** [review]
LGTM

**Comment by jithunnair-amd** [comment]
@atalman @kit1980 @malfet Can one of you please approve running CI on this PR? Thanks!

**Comment by malfet** [comment]
@jithunnair-amd done

**Comment by jithunnair-amd** [comment]
@BLOrange-AMD Since ROCm5.4.1 will have some perf updates, let's update this PR to use ROCm5.4.1 for nightly wheels.

**Comment by jithunnair-amd** [comment]
@BLOrange-AMD Let's hold this PR until we get the scripts refactor PR merged in: https://github.com/pytorch/builder/pull/1232

**Comment by BLOrange-AMD** [comment]
@atalman @kit1980 @malfet Could you please approve running CI on this PR? Thanks!

**Comment by jeffdaily** [comment]
We should move to rocm 5.4.2 since it is the latest available now.

**Comment by BLOrange-AMD** [comment]
@atalman @kit1980 @malfet Could you please approve running CI on this PR? Thanks!

**Comment by atalman** [comment]
@jithunnair-amd could you please open test PR on pytorch/pytorch to test these changes ?

**Comment by jithunnair-amd** [comment]
@atalman We filed a PR 93090 to test this PR, but IIUC, we can only test the manywheel/libtorch package builds and tests through that PR. We still need this PR to build the manywheel/libtorch docker images? The last run failed for ROCm because the base docker images for ROCm4.5.2 weren't available: `rocm/dev-centos-7:5.4.2` and `rocm/dev-ubuntu-20.04:5.4.2`. We have published those images now, so can you please retrigger the docker build jobs? Also, is there a plan to have bot commands for builder similar to pytorch repo?

**Comment by jithunnair-amd** [comment]
@BLOrange-AMD Let's wait for https://github.com/pytorch/builder/pull/1288 to be merged so that the diskspace issue is resolved first.

**Comment by jithunnair-amd** [comment]
@BLOrange-AMD https://github.com/pytorch/builder/pull/1288 has been merged

File: common/install_miopen.sh
Comment (jithunnair-amd) [inline common/install_miopen.sh:65]: @BLOrange-AMD Syntax error here: should be `elif` instead of `else if`

Comment (BLOrange-AMD) [reply common/install_miopen.sh]: Just fixed.

**Comment by BLOrange-AMD** [comment]
@atalman @kit1980 @malfet Could you please approve running CI on this PR for the latest updates? Thanks!

**Comment by jeffdaily** [comment]
@kit1980 can you restart the CI jobs?  We hit the ubuntu 18 brownout.  https://github.com/actions/runner-images/issues/6002

**Comment by atalman** [comment]
> @kit1980 can you restart the CI jobs? We hit the ubuntu 18 brownout. [actions/runner-images#6002](https://github.com/actions/runner-images/issues/6002)

restarted the CI jobs

**Comment by BLOrange-AMD** [comment]
> @BLOrange-AMD could you please open test PR on pytorch/pytorch to test these changes ? Here is one example of such PR: [pytorch/pytorch#92629](https://github.com/pytorch/pytorch/pull/92629)

Here is the PR according to the suggestion: https://github.com/pytorch/pytorch/pull/93090

**Review by atalman — APPROVED** [review]
LGTM

**Pull request merged**
